coralproject/talk
# Feat/corl 2474 allocate site mods

## What does this PR do?
This PR enables Organization Moderators to allocate Site Moderators to sites within their organization.

## What changes to the GraphQL/Database Schema does this PR introduce?
This updates the `UpdateUserRoleInput` type to accept an optional array of `siteIDs`.

## Does this PR introduce any new environment variables or feature flags? 
No.

## If any indexes were added, were they added to `INDEXES.md`?
n/a

## How do I test this PR?
1. As an organization moderator navigate to the Community view. You should see that UserRoleChange buttons are available for all users with roles <= moderator (so not for org mods or admins).
2. Click on the Role name of a non-moderator user. You should see that only the role Site Moderator is enabled.
3. Click on this role, and you should be able to add sites to this user in the following modal. Completing the modal should change that user's role to Site Moderator. 
4. Now that the user is a site mod, the only roles available to you in the RoleChange component should be Site Moderator and Commenter (this should also be true for users that were already Site Moderators when you arrived).
5. Set a site moderator back to a Commenter.
6. Log in as any other role and see that managing user roles from the Community view works (or doesn't work) just as before.
 
 
## How do we deploy this PR?
No special considerations should be required.
